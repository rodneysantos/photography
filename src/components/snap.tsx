import { Photo } from "../models";

interface SnapProps {
	// The URL of the image.
	photo: Photo;

	// The alt text of the image.
	alt: string;

	// The function that is called when the image is clicked.
	onClick: (photo: Photo) => void;
}

export const Snap = (props: SnapProps) => {
	function handleClick() {
		props.onClick(props.photo);
	}

	return (
		<img
			class="object-cover object-center w-full h-full aspect-square"
			loading="lazy"
			src={props.photo.url}
			alt={props.alt}
			onClick={handleClick}
		/>
	);
};
