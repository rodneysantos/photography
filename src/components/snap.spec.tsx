import { cleanup, render } from "@solidjs/testing-library";
import { afterEach, describe, expect, it, vi } from "vitest";
import { Snap } from "./snap";

describe("snap", () => {
	const onClickMock = vi.fn();
	const photo = {
		id: "1",
		url: "image.jpg",
		title: "photo",
	};

	afterEach(cleanup);

	afterEach(() => {
		vi.restoreAllMocks();
	});

	it("matches snapshot", () => {
		// act
		const { asFragment } = render(() => (
			<Snap photo={photo} alt="photo" onClick={onClickMock} />
		));

		// assert
		expect(asFragment()).toMatchSnapshot();
	});

	it("calls onClick when clicked", () => {
		// act
		const { getByAltText } = render(() => (
			<Snap photo={photo} alt="photo" onClick={onClickMock} />
		));
		getByAltText("photo").click();

		// assert
		expect(onClickMock).toHaveBeenCalledWith(photo);
	});
});
