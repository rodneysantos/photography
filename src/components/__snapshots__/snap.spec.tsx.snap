// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snap > matches snapshot 1`] = `"<img class=\\"object-cover object-center w-full h-full\\" loading=\\"lazy\\" src=\\"image.jpg\\" alt=\\"photo\\">"`;
